doc/sphinx/addendum/ring.rst

```diff
@@ -653,18 +659,21 @@ zero for the correctness of the algorithm.
 
 The syntax for adding a new field is 
 
-.. cmd:: Add Field @ident : @term {? ( @field_mod {* , @field_mod } )}
+.. cmd:: @ident : @term1_extended {? ( {+, @field_mod } ) }
+
+   .. insertprodn field_mod field_mod
 
-   The :n:`@ident` is not relevant. It is used just for error
-   messages. :n:`@term` is a proof that the field signature satisfies the
+   .. prodn::
+      field_mod ::= @ring_mod
+      | completeness @term1_extended
+
+   The :n:`@ident` is used only for error
+   messages. :n:`@term1_extended` is a proof that the field signature satisfies the
    (semi-)field axioms. The optional list of modifiers is used to tailor
    the behavior of the tactic.
 
-   .. productionlist:: coq
-      field_mod : `ring_mod` | completeness `term`
-
    Since field tactics are built upon ``ring``
-   tactics, all modifiers of the ``Add Ring`` apply. There is only one
+   tactics, all modifiers of :cmd:`Add Ring` apply. There is only one
    specific modifier:
 
    completeness :n:`@term`
```

Comment:
In the mlg files it is "constr".  I think we picked this name to emphasize the relationship to terms. 
 There are 233 references to it in orderedGrammar.  I'm open to suggestions.

```
term1_extended: [
| term1
| "@" qualid OPT univ_annot
]
```

Reply:
OK, thank you for your answer. After looking at the references in `orderedGrammar`, I now understand that the whole point of this distinction between `constr` and `lconstr` is that using `constr` allows requiring a list of terms separated by spaces (or other things that could be confused with terms, separated by spaces) without confusing this with function application. If you want to pass complex terms where a `constr` is expected, you have to put parentheses. `lconstr` can be used only where not putting parentheses would not risk creating ambiguity with the surroundings.

Given the (frequent) use case, we should move its definition to the main grammar chunk defining terms and give it a more intelligible name.

After thinking about this for less than an hour, I am currently leaning toward `single_term` as this would express the fact that if you want to pass a single term that could be confused with a space-separated list of terms, then you have to put parentheses (in fact most tactics do expect this).

Another option would be `parenthesized_term`, which reminds the reader that most of the time, you need to put surrounding parentheses. But it's not really satisfying since that is not *always* the case.

If you have a better idea, I'm all hears.

Reply:
Since this symbol is used so much, maybe we could use something shorter and a little less readable, plus a note where it's defined about the need for parentheses.  Some stuff the user just has to learn.  Maybe `a_term`?

And maybe move the definition to just below the definition for `term1` in the Terms section.

Reply:
>  Maybe `a_term`?

What about `one_term`? A little less short but also less cryptic.

> And maybe move the definition to just below the definition for `term1` in the Terms section.

Just below, or just above? We should also insert an explanation so that users do not have to take time to grasp why it's defined. Something like the following:

Lots of commands and tactics expect `one_term` instead of `term`. What that means is that terms will have to be surrounded by parentheses unless they are very simple (such as a single identifier). This avoids confusing a space-separated list of terms with a term applied to its arguments.

Reply:
FWIW, there are 233 uses of `one_term`, but only 20 of them are in the form `LIST* one_term`.
But I suppose `Derive @ident SuchThat @one_term As @ident` could confuse the parser for some terms when it gets to "As", which is not a keyword.  (I think it should be lower case "as" here, BTW).

Reply:
Exactly! Regarding the casing, a bit too late and not important enough to change IMHO.

Reply:
Should be `one_term`